app/components/RelatedRecordTable.tsx

```diff
@@ -4,12 +4,54 @@ import RelatedRecordTableRow from "./RelatedRecordTableRow";
 import { ColDataCardDef } from "@/types/types";
 
 interface RelatedRecordTableProps<T extends Record<string, unknown>> {
+  /**
+   * The records to be rendered in the table
+   */
   records: T[];
+  /**
+   * The table's column definitions
+   */
   columns: ColDataCardDef<T>[];
+  /**
+   * Graphql mutation that will be exectuted when a row is edited -
+   * will also be passed to the rowActionButton, if present
+   */
   mutation: string;
+  /**
+   * If the SWR refetcher is (re)validating
+   */
   isValidating: boolean;
+  /**
+   * The title to be rendered in the table's card header
+   */
   title: string;
+  /**
+   * Optional React component to be rendered in the card's header
+   */
   headerActionButton?: React.ReactNode;
+  /**
+   * Optional react component to be rendered in the rightmost
+   * column of every row
+   */
+  rowActionButton?: React.ComponentType<RowActionButtonProps<T>>;
+  /**
+   * Callback function to be executed after a row edit is saved
+   */
+  onSaveCallback: () => Promise<void>;
+}
+
+export interface RowActionButtonProps<T extends Record<string, unknown>> {
```

Comment:
as you can see, I copied the `HeaderActionButton` for this. i can foresee a few potential improvements, which might be worth implementing now:

- Ability to pass a column header label. Right now the column header is blank
- Ability to have multiple row action buttons by passing a providing an array of components

Reply:
i think this is great, and probably fine to wait to add those other improvements as needed?